src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -641,12 +657,7 @@ protected SymbolAndDiagnostics FindImplementationForInterfaceMemberWithDiagnosti
                         return SymbolAndDiagnostics.Empty;
                     }
 
-                    if (info.implementationForInterfaceMemberMap == null)
-                    {
-                        Interlocked.CompareExchange(ref info.implementationForInterfaceMemberMap, new ConcurrentDictionary<Symbol, SymbolAndDiagnostics>(), null);
-                    }
-
-                    return info.implementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
+                    return info.ImplementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
```

Comment:
Did Roslyn implement support for caching delegates when passing in a method group?  If not, isn't the second parameter always resulting in a delegate allocation?